src/main/java/org/spongepowered/common/item/inventory/lens/impl/MinecraftLens.java

```diff
@@ -31,11 +31,25 @@
 import org.spongepowered.common.item.inventory.lens.Fabric;
 import org.spongepowered.common.item.inventory.lens.Lens;
 import org.spongepowered.common.item.inventory.lens.SlotProvider;
+import org.spongepowered.common.item.inventory.lens.impl.collections.SlotCollection;
 
 import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
 
 public abstract class MinecraftLens extends AbstractLens<IInventory, ItemStack> {
 
+    private static Map<Class, Map<Integer, ReusableLens>> reusableLenses = new HashMap<>();
```

Comment:
Is it worth using a special map from fastutil?
`private static final Map<Class, Int2ObjectMap<ReusableLens>> reusableLenses = new HashMap<>();`
...
`k -> new Int2ObjectOpenHashMap<>()`